include/effects/matrix/PatternWeather.h

```diff
@@ -165,75 +238,113 @@ class PatternWeather : public LEDStripEffect
+                strCountryCode = configCountryCode;
+                locationValid = true;
+            }
+            else
+            {
+                debugW("Bad location JSON: %s", de.c_str());
+            }
+
+        }
+        else
         {
             debugW("Error fetching coordinates for location: %s", configLocation.c_str());
-            http.end();
             return false;
         }
 
-        AllocatedJsonDocument doc(4096);
-        deserializeJson(doc, http.getString());
-        JsonObject coordinates = configLocationIsZip ? doc.as<JsonObject>() : doc[0].as<JsonObject>();
-
-        strLatitude = coordinates["lat"].as<String>();
-        strLongitude = coordinates["lon"].as<String>();
-
         http.end();
-
-        strLocation = configLocation;
-        strCountryCode = configCountryCode;
-
-        return true;
+        return locationValid;
     }
 
-    // getTomorrowTemps
-    //
-    // Request a forecast and then parse out the high and low temps for tomorrow
-
-    bool getTomorrowTemps(float& highTemp, float& lowTemp)
+    /**
+     * @brief Get the Tomorrow Temperatures
+     * 
+     * Request a forecast and then parse out the high and low temps for tomorrow
+     * along with the weather icon for mid-day
+     * 
+     * @param highTemp
+     * @param lowTemp
+     * @param icon 
+     * @return true Successful data retrieval
+     * @return false Failed data retrieval
+     */
+    bool getTomorrowTemps(float& highTemp, float& lowTemp, int& icon)
     {
         HTTPClient http;
         String url = "http://api.openweathermap.org/data/2.5/forecast"
-            "?lat=" + strLatitude + "&lon=" + strLongitude + "&appid=" + urlEncode(g_ptrSystem->DeviceConfig().GetOpenWeatherAPIKey());
+            "?lat=" + strLatitude + "&lon=" + strLongitude + "&cnt=18&appid=" + urlEncode(g_ptrSystem->DeviceConfig().GetOpenWeatherAPIKey());
+        debugI("tomorrowURL: %s", url.c_str());
         http.begin(url);
         int httpResponseCode = http.GET();
 
         if (httpResponseCode > 0)
         {
-            AllocatedJsonDocument doc(4096);
-            deserializeJson(doc, http.getString());
+            AllocatedJsonDocument doc(9216);
+            String tomorrowJson = http.getString();
+            DeserializationError de = deserializeJson(doc, tomorrowJson);
+            String count = doc["cnt"];
+            debugI("Return count %s DE code %s", count.c_str(), de.c_str());
             JsonArray list = doc["list"];
 
-            // Get tomorrow's date
-            time_t tomorrow = time(nullptr) + 86400;
-            tm* tomorrowTime = localtime(&tomorrow);
-            char dateStr[11];
-            strftime(dateStr, sizeof(dateStr), "%Y-%m-%d", tomorrowTime);
-
-            iconTomorrow = -1;
-
-            // Look for the temperature data for tomorrow
-            for (size_t i = 0; i < list.size(); ++i)
+            // Get tomorrow's date 
+            time_t tomorrow = time(nullptr) + 86400; 

```

Comment:
You are more versed in the changes on C++ than I am.  It has been too long since I have had to know any of that at any level of detail.  I only know the old C-style as that is what I learned back in the early 80's.  hmm. this comment didn't post when I made it last night.